File: app/src/main/java/com/utynote/app/AppRoot.java
package com.utynote.app;


import android.app.Application;
import android.support.annotation.NonNull;

import com.utynote.app.dependencies.AppModule;
import com.utynote.app.dependencies.LibModule;
import com.utynote.app.dependencies.NetModule;
import com.utynote.app.dependencies.search.DaggerSearchComponent;
import com.utynote.app.dependencies.search.SearchComponent;
import com.utynote.app.dependencies.search.SearchModule;

import static com.utynote.utils.Preconditions.checkNotNull;

public class AppRoot extends Application {
    private SearchComponent searchComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        AppConfig config = new AppConfig.Builder()
                .withSearch("https://search.mapzen.com/v1/", "geojson")
                .build();

        searchComponent = DaggerSearchComponent.builder()
                .appModule(new AppModule(this))
                .netModule(new NetModule())
                .libModule(new LibModule())
                .searchModule(new SearchModule(config.search))
                .build();
    }

    @NonNull
    public SearchComponent getSearchComponent() {
        return checkNotNull(searchComponent);
    }
}

File: app/src/main/java/com/utynote/app/AppRoot.kt
package com.utynote.app


import android.app.Application

import com.utynote.app.dependencies.AppModule
import com.utynote.app.dependencies.LibModule
import com.utynote.app.dependencies.NetModule
import com.utynote.app.dependencies.search.DaggerSearchComponent
import com.utynote.app.dependencies.search.SearchComponent
import com.utynote.app.dependencies.search.SearchModule


class AppRoot : Application() {
    private var searchComponent: SearchComponent? = null

    override fun onCreate() {
        super.onCreate()

        val config = AppConfig(AppConfig.Search(
                        server = "https://search.mapzen.com/v1/",
                        format = "geojson"))

        searchComponent = DaggerSearchComponent.builder()
                .appModule(AppModule(this))
                .netModule(NetModule())
                .libModule(LibModule())
                .searchModule(SearchModule(config.search))
                .build()
    }

    fun getSearchComponent(): SearchComponent {
        return checkNotNull(searchComponent)
    }
}
